Makefile

```diff
@@ -29,7 +29,7 @@ codegen: models services examples init-files
 check-codegen: ## Check if the code is up to date
 check-codegen:
 	@$(MAKE) codegen
-	@git diff --exit-code || echo "Code is not up-to-date. Please run 'make codegen'"
+	@git diff --exit-code || echo "Code is not up-to-date. Please run 'make codegen'" && exit 1
```

Comment:
Changed this so we stop missing non-regenerated examples in CI e.g. for #1108 - https://github.com/argoproj-labs/hera/actions/runs/9566065952/job/26370421288 - had the `Code is not up-to-date. Please run 'make codegen'` message but no failure

Reply:
Moved to #1110